js-packages/@prestojs/ui-antd/src/AntdUiProvider.tsx

```diff
@@ -0,0 +1,200 @@
+import { getFormatterForField, UiProvider, UiProviderProps } from '@prestojs/ui';
+
+import type {
+    PickerBaseProps,
+    PickerDateProps,
+    PickerTimeProps,
+} from 'antd/es/date-picker/generatePicker';
+import type { ComponentClass, ForwardRefExoticComponent, LazyExoticComponent } from 'react';
+import React, { useContext, useMemo } from 'react';
+import FormItemWrapper from './FormItemWrapper';
+import FormWrapper from './FormWrapper';
+import getWidgetForField from './getWidgetForField';
+
+type _DatePickerComponent<T> = ComponentClass<
+    PickerBaseProps<T> | PickerDateProps<T> | PickerTimeProps<T>,
+    any
+>;
+type DatePickerComponent<T> =
+    | _DatePickerComponent<T>
+    | LazyExoticComponent<_DatePickerComponent<T>>;
+
+type _TimePickerComponent<T> = ForwardRefExoticComponent<
+    Omit<PickerTimeProps<T>, 'picker'> & React.RefAttributes<any>
+>;
+type TimePickerComponent<T> =
+    | _TimePickerComponent<T>
+    | LazyExoticComponent<_TimePickerComponent<T>>;
+
+interface AntdUiConfigComponents {
+    DatePicker?: DatePickerComponent<any>;
+    TimePicker?: TimePickerComponent<any>;
+}
+
+/**
+ * Class to store current antd config created by [AntdUiProvider](doc:AntdUiProvider).
+ *
+ * To get the current config see [useAntdUiConfig](doc:useAntdUiConfig).
+ *
+ * @extract-docs
+ */
+export class AntdUiConfig {
+    components: AntdUiConfigComponents;
+
+    constructor(components: AntdUiConfigComponents) {
+        this.components = components;
+    }
+
+    /**
+     * Get the `DatePicker` component to use. This is used by all date/datetime based widgets (eg. [DateWidget](doc:DateWidget])).
+     */
+    getDatePicker<T>(): DatePickerComponent<T> {
+        if (!this.components.DatePicker) {
+            throw new Error(`To use components that use DatePicker you must first configure @prestojs/ui-antd so it knows the type to use. To set the default moment based datepicker add the following to your app before any widgets are used:
+
+    import { DatePicker } from 'antd';
+
+    <AntdUiProvider datePickerComponent={DatePicker}>...</AntdUiProvider>
+
+See https://prestojs.com/docs/ui-antd/ for instructions on setting up @prestojs/ui-antd and 
+https://ant.design/docs/react/replace-moment#DatePicker for options in customising the DatePicker
+`);
```

Comment:
i think we should mention that moment is now legacy and generally discouraged
(ant's not doing that for major version backwards compatibility reason but we dont have that burden)